state_transfer/labels/labels.go

```diff
@@ -1,3 +1,25 @@
 package labels
 
+import (
+	"fmt"
+
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
 var Labels = map[string]string{"app": "crane2"}
+
+func ValidateLabels(labels map[string]string) (err error) {
+	var errs []error
+	for key, val := range labels {
+		err := validation.IsDNS1123Label(key)
```

Comment:
do labels need to be DNS1123 compliant? 

```
Labels are key/value pairs. Valid label keys have two segments: an optional prefix and name, separated by a slash (/). The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between. The prefix is optional. If specified, the prefix must be a DNS subdomain: a series of DNS labels separated by dots (.), not longer than 253 characters in total, followed by a slash (/).
```

I think this is a little different than the DNS check but maybe I am incorrect? Just something to look at.

Reply:
Thanks for pointing out - DNS label here probably means just the DNS label present between two dots in a DNS address. Need to change this validation function.